File: frontend/src/wails/wails_runtime.ts
import {WailsRuntimeApi} from "@/wails/wails_runtime_api";
import {EventsOn} from "@wails/runtime";

export class WailsRuntime implements WailsRuntimeApi {
    EventsOn(eventName: string, callback: (...data: any) => void): () => void {
        return EventsOn(eventName, callback);
    }
}

File: internal/interfaces/wails_runtime.go
package interfaces

import (
	"github.com/wailsapp/wails/v2/pkg/runtime"
)

type WailsRuntime interface {
	OpenFileDialog(dialogOptions runtime.OpenDialogOptions) (string, error)
	SaveFileDialog(dialogOptions runtime.SaveDialogOptions) (string, error)
	EventsEmit(eventName string, optionalData ...any)
}
